source/dhtslib/vcf/record.d

```diff
@@ -459,8 +711,12 @@ struct VCFRecord
     /// "If flt_id is PASS, all existing filters are removed first. If other than PASS, existing PASS is removed."
     int addFilter(string f)
     {
-        return bcf_add_filter(this.vcfheader.hdr, this.line, 
-            bcf_hdr_id2int(this.vcfheader.hdr, BCF_DT_ID, toStringz(f)));
+        
+        if(f == "") hts_log_warning(__FUNCTION__, "filter was empty");
+        const int fid = bcf_hdr_id2int(this.vcfheader.hdr, BCF_DT_ID, toStringz(f));
+        if (fid == -1) hts_log_warning(__FUNCTION__, format("filter not found in header (ignoring): %s", f) );
```

Comment:
Probably, I tried not to alter previous methods too much. This was an older method.

Reply:
It looks like setting a filter of -1 might cause undefined behavior since they never check for this. I would make the function return early if the filter was not found (i.e. after the warning above)

```C
int bcf_add_filter(const bcf_hdr_t *hdr, bcf1_t *line, int flt_id)
{
    if ( !(line->unpacked & BCF_UN_FLT) ) bcf_unpack(line, BCF_UN_FLT);
    int i;
    for (i=0; i<line->d.n_flt; i++)
        if ( flt_id==line->d.flt[i] ) break;
    if ( i<line->d.n_flt ) return 0;    // this filter is already set
    line->d.shared_dirty |= BCF1_DIRTY_FLT;
    if ( flt_id==0 )    // set to PASS
        line->d.n_flt = 1;
    else if ( line->d.n_flt==1 && line->d.flt[0]==0 )
        line->d.n_flt = 1;
    else
        line->d.n_flt++;
    hts_expand(int, line->d.n_flt, line->d.m_flt, line->d.flt);
    line->d.flt[line->d.n_flt-1] = flt_id;
    return 1;
}
```

Reply:
addressed in
0910089